pkg/common/defaulter/trafficmanagerprofile.go

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+// Package defaulter provides the utils for setting default values for a resource.
+package defaulter
+
+import (
+	"k8s.io/utils/ptr"
+
+	fleetnetv1alpha1 "go.goms.io/fleet-networking/api/v1alpha1"
+)
+
+// SetDefaultsTrafficManagerProfile sets the default values for TrafficManagerProfile.
+func SetDefaultsTrafficManagerProfile(obj *fleetnetv1alpha1.TrafficManagerProfile) {
+	if obj.Spec.MonitorConfig == nil {
+		obj.Spec.MonitorConfig = &fleetnetv1alpha1.MonitorConfig{}
+	}
+
+	if obj.Spec.MonitorConfig.IntervalInSeconds == nil {
+		obj.Spec.MonitorConfig.IntervalInSeconds = ptr.To(int64(30))
+	}
+
+	if obj.Spec.MonitorConfig.Path == nil {
+		obj.Spec.MonitorConfig.Path = ptr.To("/")
+	}
+
+	if obj.Spec.MonitorConfig.Port == nil {
+		obj.Spec.MonitorConfig.Port = ptr.To(int64(80))
+	}
+
+	if obj.Spec.MonitorConfig.Protocol == nil {
+		obj.Spec.MonitorConfig.Protocol = ptr.To(fleetnetv1alpha1.TrafficManagerMonitorProtocolHTTP)
+	}
+
+	// TimeoutInSeconds value depends on the IntervalInSeconds, so that the defaulter MUST handle the IntervalInSeconds first.
+	// * If the Probing Interval is set to 30 seconds, then you can set the Timeout value between 5 and 10 seconds.
+	//   If no value is specified, it uses a default value of 10 seconds.
+	// * If the Probing Interval is set to 10 seconds, then you can set the Timeout value between 5 and 9 seconds.
+	//   If no Timeout value is specified, it uses a default value of 9 seconds.
+	// Reference link: https://learn.microsoft.com/en-us/azure/traffic-manager/traffic-manager-monitoring#configure-endpoint-monitoring
+	if obj.Spec.MonitorConfig.TimeoutInSeconds == nil {
+		if *obj.Spec.MonitorConfig.IntervalInSeconds == 30 {
```

Comment:
what if the internal in seconds is more than 30?

Reply:
You can specify two values here: 30 seconds (normal probing) and 10 seconds (fast probing) for the intervalInSeconds.

My bad, i thought validation webhook would be invoked first. Will fix it in my next PR.